controllers/operatorconfig_controller.go

```diff
@@ -54,51 +71,341 @@ type OperatorConfigReconciler struct {
+	}
+
+	log.Info("All needed resources are present")
+
+	return ctrl.Result{}, nil
+}
+
+func (r *OperatorConfigReconciler) CleanUpOpjects(ctx context.Context) error {
+	filepaths := []string{"proxy.yaml", "crane-ui-plugin.yaml", "manifests.yaml"}
+
+	for _, f := range filepaths {
+		err := r.deleteResourcesFromFile(f, ctx)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *OperatorConfigReconciler) createResourcesFromFile(path string, ctx context.Context, log logr.Logger, operatorConfig *cranev1alpha1.OperatorConfig) error {
 
-	err := createClusterTasks(ctx, log)
+	var decoder = yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
 
+	data, err := getResources(path)
 	if err != nil {
-		log.Error(err, "Error creating cluster tasks")
-		return ctrl.Result{Requeue: true}, nil
-	} else {
-		log.Info("All the needed cluster task created")
+		return err
+	}
+
+	for _, resource := range data {
+		if len(resource) > 0 {
+			obj := unstructured.Unstructured{}
+			_, gvk, err := decoder.Decode([]byte(resource), nil, &obj)
+			if err != nil {
+				return err
+			}
+
+			switch gvk.Kind {
+			case "Deployment":
+				err = r.createDeployment(&obj, ctx, log, operatorConfig)
+				if err != nil {
+					return err
+				}
+			case "Service":
+				err = r.createService(&obj, ctx, log, operatorConfig)
+				if err != nil {
+					return err
+				}
+			case "ConfigMap":
+				err = r.createConfigMap(&obj, ctx, log, operatorConfig)
+				if err != nil {
+					return err
+				}
+			case "ClusterTask":
+				err = r.createClusterTask(&obj, ctx, log, operatorConfig)
+				if err != nil {
+					return err
+				}
+			case "ConsolePlugin":
+
+				err = r.createConsolePlugin(&obj, ctx, log, operatorConfig)
+				if err != nil {
+					return err
+				}
+			}
+		}
 	}
 
-	err = configureCranePlugin(ctx, log)
+	return nil
+}
 
+func (r *OperatorConfigReconciler) createDeployment(resource *unstructured.Unstructured, ctx context.Context, log logr.Logger, oc *cranev1alpha1.OperatorConfig) error {
+
+	var obj v1.Deployment
+	err := runtime.DefaultUnstructuredConverter.
+		FromUnstructured(resource.UnstructuredContent(), &obj)
 	if err != nil {
-		log.Error(err, "error configuring crane plugin")
-		return ctrl.Result{Requeue: true}, nil
-	} else {
-		log.Info("Crane ui plugin configured")
+		return err
 	}
 
-	err = configureProxy(ctx, log)
+	err = controllerutil.SetControllerReference(oc, &obj, r.Scheme)
 	if err != nil {
-		log.Error(err, "error configuring proxy")
-		return ctrl.Result{Requeue: true}, nil
-	} else {
-		log.Info("Proxy Configured")
+		return err
 	}
+	log.Info(fmt.Sprintf("creating %s/%s", obj.Kind, obj.Name))
+	if err := r.Create(ctx, &obj); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			log.Error(err, fmt.Sprintf("Failed to create %s", obj.Kind), "Namespace", obj.Namespace, "Name", obj.Name)
+			return err
+		}
+	}
+
+		tmp := v1.Deployment{}
+		if err := wait.Poll(time.Second*1, time.Second*15, func() (done bool, err error) {
```

Comment:
it is unusual to have polling inside of the control loop like that. Would like to understand problem is being solved by polling here?

Depending on the problem, I would recommend, removing the poll and relying on the controller-runtime watch events to trigger the recons

Reply:
Sometimes we might hit `notExists` error on immediate `Get` after `Create`. To make sure that doesn't happen I opted for poll.

Reply:
If you get error, just return with a request should work right?

1. Create
2. Cache is late
3. Get will miss, you return error
4.  Requeue
5. Enter reconciliation again 
6.  Get will succeed